client/src/main/java/com/arcbees/beestore/client/application/SessionStorageHandler.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2015 ArcBees Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.arcbees.beestore.client.application;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.google.gwt.user.client.rpc.SerializationException;
+import com.google.inject.Inject;
+import com.seanchenxi.gwt.storage.client.StorageExt;
+import com.seanchenxi.gwt.storage.client.StorageKey;
+import com.seanchenxi.gwt.storage.client.StorageKeyFactory;
+
+public class SessionStorageHandler {
```

Comment:
You should use LocalStorage instead. SessionStorage is "per tab", it works when you reload, but not when you open a link in a new tab. See http://stackoverflow.com/a/5523174